ambry-router/src/main/java/com/github/ambry/router/GetOperation.java

```diff
@@ -45,7 +44,7 @@ abstract class GetOperation {
   protected final NonBlockingRouterMetrics routerMetrics;
   protected final ClusterMap clusterMap;
   protected final ResponseHandler responseHandler;
-  protected final Callback<GetBlobResultInternal> getOperationCallback;
+  protected final Callback<GetBlobResult> getOperationCallback;
```

Comment:
Are any of these changes really needed to add a new request type to get BlobChunkIds ? The PR is expanding and its getting harder to rread

Reply:
It's not a new type, It's the existing type which we should use. Previously we use GetBlobResultInternal is because we don't have blobChunkIds in GetBlobResult. With my new changes, we should use GetBlobResult.